### PR TITLE
Create gitignore file and basic structure

### PR DESCRIPTION
**Pull request by adeptofvoltron**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+# TeXlipse plugin
+.texlipse
+
+# STS (Spring Tool Suite)
+.springBeans
+
+# Code Recommenders
+.recommenders/
+
+# Scala IDE specific (Scala & Java development for Eclipse)
```
Comment (pradzikowski) [inline .gitignore]: I think we dont need it ....

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+# sbteclipse plugin
+.target
+
+# Tern plugin
+.tern-project
+
+# TeXlipse plugin
+.texlipse
+
+# STS (Spring Tool Suite)
```
Comment (pradzikowski) [inline .gitignore]: STS is java thing...

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+# PyDev specific (Python IDE for Eclipse)
+*.pydevproject
+
+# CDT-specific (C/C++ Development Tooling)
+.cproject
+
+# JDT-specific (Eclipse Java Development Tools)
+.classpath
+
+# Java annotation processor (APT)
```
Comment (pradzikowski) [inline .gitignore]: Le foo Java !

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+# Eclipse Core
+.project
+
+# External tool builders
+.externalToolBuilders/
+
+# Locally stored "Eclipse launch configurations"
+*.launch
+
+# PyDev specific (Python IDE for Eclipse)
```
Comment (pradzikowski) [inline .gitignore]: Python @.@

File: composer.json [hunk 5]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "purring_cat/valueobject",
+  "description": "todo",
+  "type": "metaapackage",
```
Comment (pradzikowski) [inline composer.json]: To much 'a' in this type :)

File: composer.json [hunk 6]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "purring_cat/valueobject",
```
Comment (pradzikowski) [inline composer.json]: Imo Name should be the same with namespace. Remember - name is a packages title on packagist.

File: src/ValueObjectInterface.php [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+declare(strict_types=1);
+
+namespace PCF\ValueObject;
+
+interface ValueObjectInterface
+{
+    /**
+     * Two value objects are equal if their values are equal
+     * @param ValueObjectInterface $compare
+     * @return boolean
```
Comment (pradzikowski) [inline src/ValueObjectInterface.php]: i suggest add line before return

File: src/ValueObjectInterface.php [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+     */
+    public function isEqualTo(ValueObjectInterface $compare): bool ;
+
+    /**
+     * Return string representation of value
+     * @return string
+     */
+    public function __toString(): string;
+}
```
Comment (pradzikowski) [inline src/ValueObjectInterface.php]: One line after bracket

File: src/ValueObjectInterface.php [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+    /**
+     * Two value objects are equal if their values are equal
+     * @param ValueObjectInterface $compare
+     * @return boolean
+     */
+    public function isEqualTo(ValueObjectInterface $compare): bool ;
+
+    /**
+     * Return string representation of value
+     * @return string
```
Comment (pradzikowski) [inline src/ValueObjectInterface.php]: Line before return

**Review by pradzikowski — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 4 above)
Comment (adeptofvoltron) [reply .gitignore]: just .gitignore plugin template. does not have any negative affect atm.

File: .gitignore (hunk 3 above)
Comment (adeptofvoltron) [reply .gitignore]: just .gitignore plugin template. does not have any negative affect atm.

File: .gitignore (hunk 2 above)
Comment (adeptofvoltron) [reply .gitignore]: just .gitignore plugin template. does not have any negative affect atm.

File: .gitignore (hunk 1 above)
Comment (adeptofvoltron) [reply .gitignore]: just .gitignore plugin template. does not have any negative affect atm.

File: .gitignore (hunk 4 above)
Comment (adeptofvoltron) [reply .gitignore]: deleted

File: .gitignore (hunk 3 above)
Comment (adeptofvoltron) [reply .gitignore]: deleted

File: .gitignore (hunk 2 above)
Comment (adeptofvoltron) [reply .gitignore]: deleted

File: .gitignore (hunk 1 above)
Comment (adeptofvoltron) [reply .gitignore]: deleted

File: composer.json (hunk 6 above)
Comment (adeptofvoltron) [reply composer.json]: done

File: src/ValueObjectInterface.php [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+
+interface ValueObjectInterface
+{
+    /**
+     * Two value objects are equal if their values are equal
+     * @param ValueObjectInterface $compare
+     *
+     * @return boolean
+     */
+    public function isEqualTo(ValueObjectInterface $compare): bool ;
```
Comment (pradzikowski) [inline src/ValueObjectInterface.php]: Space before semicolon

File: src/ValueObjectInterface.php (hunk 8 above)
Comment (adeptofvoltron) [reply src/ValueObjectInterface.php]: done - u [nazi programmer](https://i.imgflip.com/ioemm.jpg)!!

File: .gitignore [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+# Eclipse Core
+.project
+
+# External tool builders
+.externalToolBuilders/
+
+# Locally stored "Eclipse launch configurations"
+*.launch
+
+# CDT-specific (C/C++ Development Tooling)
```
Comment (pradzikowski) [inline .gitignore:48]: Yo forget cpp ? :D

File: src/ValueObjectInterface.php (hunk 9 above)
Comment (adeptofvoltron) [reply src/ValueObjectInterface.php]: done

File: src/ValueObjectInterface.php (hunk 7 above)
Comment (adeptofvoltron) [reply src/ValueObjectInterface.php]: done

File: composer.json (hunk 5 above)
Comment (adeptofvoltron) [reply composer.json]: done

File: .gitignore (hunk 11 above)
Comment (adeptofvoltron) [reply .gitignore:48]: YAGNI - just a template stuff ;)

File: src/ValueObjectInterface.php (hunk 10 above)
Comment (adeptofvoltron) [reply src/ValueObjectInterface.php]: done

**Review by pradzikowski — APPROVED** [review]

**Comment by pradzikowski** [comment]
image

**Pull request merged**